### PR TITLE
Add 'phone-number' exercise

### PR DESCRIPTION
**Pull request by palisaide**
See the exercise description [here](https://github.com/exercism/x-common/blob/master/exercises/phone-number/description.md)
Test cases were copied from the [C++ track](https://github.com/exercism/xcpp/blob/master/phone-number/phone_number_test.cpp) as there is no canonical-data.json for this exercise.
- [x] Set up makefile, test file, and implementation files
- [x] Add tests
- [x] Add example

### CODE REVIEW

**Comment by palisaide** [comment]
I'm not particularly happy with the implementation of `phone_number_format()`, so I'm open to any suggestions on how to make that one cleaner.

File: exercises/phone-number/test/test_phone_number.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#include "vendor/unity.h"
+#include "../src/phone_number.h"
+#include <stdlib.h>
+
+void test_phone_number(const char *input, const char *expected)
```
Comment (ryanplusplus) [inline exercises/phone-number/test/test_phone_number.c]: Should this be `test_phone_number_clean`?

File: exercises/phone-number/src/example.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      }
+   }
+
+   return output;
+}
+
+char *phone_number_get_area_code(const char *input)
+{
+   char *output = calloc(AREA_CODE_LENGTH, sizeof(char));
+   strncpy(output, phone_number_clean(input), AREA_CODE_LENGTH);
```
Comment (ryanplusplus) [inline exercises/phone-number/src/example.c]: A simple alternative would be to just set `phone_number_clean(input)[4] = '\0'` and return that. It uses a little bit of extra memory, but the client can still `free` it like normal.

File: exercises/phone-number/src/example.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+   *cursor++ = ' ';
+   memcpy(cursor, &input[AREA_CODE_LENGTH], EXCHANGE_LENGTH);
+   cursor += EXCHANGE_LENGTH;
+   *cursor++ = '-';
+   memcpy(cursor, &input[AREA_CODE_LENGTH + EXCHANGE_LENGTH], EXTENSION_LENGTH);
+   cursor += EXTENSION_LENGTH;
+   *cursor++ = '\0';
+
+   return output;
+}
```
Comment (ryanplusplus) [inline exercises/phone-number/src/example.c]: I'm fairly certain you can use `sprintf` for this:

```c
sprintf(output, "(%.3s) %.3s-%.4s", input, input[AREA_CODE_LENGTH], input[AREA_CODE_LENGTH + EXCHANGE_LENGTH);
```

**Comment by ryanplusplus** [comment]
Sorry, I merged some of your other PRs and created a conflict

File: exercises/phone-number/src/example.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+char *phone_number_get_area_code(const char *input)
+{
+   char *output = calloc(AREA_CODE_LENGTH, sizeof(char));
+   strncpy(output, phone_number_clean(input), AREA_CODE_LENGTH);
+
+   return output;
+}
+
+char *phone_number_format(const char *input)
```
Comment (ryanplusplus) [inline exercises/phone-number/src/example.c]: `phone_number_get_area_code` uses `phone_number_clean` internally, but this doesn't. I think we should add a test that requires this for consistency.

**Comment by patricksjackson** [comment]
I don't see anything other than what @ryanplusplus already noted. The rest LGTM 👍

File: exercises/phone-number/src/example.c (hunk 2 above)
Comment (palisaide) [inline exercises/phone-number/src/example.c]: Good idea!

File: exercises/phone-number/src/example.c (hunk 3 above)
Comment (palisaide) [inline exercises/phone-number/src/example.c]: Yes, much cleaner

File: exercises/phone-number/src/example.c (hunk 4 above)
Comment (palisaide) [inline exercises/phone-number/src/example.c]: I agree.  I'll add a test for this.

**Comment by palisaide** [comment]
Ready for another look

**Comment by ryanplusplus** [comment]
Changes look good to me 👍

**Pull request merged**